### PR TITLE
CRUD operations

### PR DESCRIPTION
**Pull request by maksimk99**
I've created 'add, update and delete' operations for car sale advertisement
For more information see #11

### CODE REVIEW

File: src/main/java/com/carhouse/controller/CarSaleController.java [hunk 1]
```diff
@@ -90,16 +90,91 @@ public String addCarSale(final Model model) {
...
+     * Return page to update new car sale advertisement.
+     * Create list of selected car feature's id to select car feature
+     * Gets the request url to return to the same page
+     *
+     * @param carSaleId  the car sale id
+     * @param requestUrl the request url
+     * @param model      model
+     * @return view string
+     */
+    @GetMapping("/carSale/{carSaleId}/update")
```
Comment (aleksieiev) [inline src/main/java/com/carhouse/controller/CarSaleController.java]: `updateCarSale` for `@GetMapping`?

File: src/main/java/com/carhouse/controller/CarSaleController.java [hunk 2]
```diff
@@ -90,16 +90,91 @@ public String addCarSale(final Model model) {
...
+     * Submit update car sale advertisement.
+     * Take list of selected car feature's id as request param
+     * Return page with list car sales
+     *
+     * @param carSale     the car sale
+     * @param requestUrl  the request url
+     * @param featureList the feature list
+     * @return the string
+     */
+    @PostMapping("/carSale/update")
```
Comment (aleksieiev) [inline src/main/java/com/carhouse/controller/CarSaleController.java]: the same like above

File: src/main/java/com/carhouse/controller/CarSaleController.java [hunk 3]
```diff
@@ -90,16 +90,91 @@ public String addCarSale(final Model model) {
...
+
+    /**
+     * Delete car sale string.
+     * Gets the request url to return to the same page
+     *
+     * @param carSaleId  the car sale id
+     * @param requestUrl the request url
+     * @return the string
+     */
+    @GetMapping("/carSale/{carSaleId}/delete")
```
Comment (aleksieiev) [inline src/main/java/com/carhouse/controller/CarSaleController.java:160]: get for delete operation?

File: src/main/java/com/carhouse/controller/CarSaleController.java (hunk 1 above)
Comment (maksimk99) [reply src/main/java/com/carhouse/controller/CarSaleController.java]: Is `getUpdateCarSalePage` better?

Comment (aleksieiev) [reply src/main/java/com/carhouse/controller/CarSaleController.java]: Good joke

Comment (aleksieiev) [reply src/main/java/com/carhouse/controller/CarSaleController.java]: 1. `@GetMapping("/carSale/{carSaleId}/updateForm")`
2. `getUpdateForm`

File: src/main/java/com/carhouse/controller/CarSaleController.java (hunk 3 above)
Comment (maksimk99) [reply src/main/java/com/carhouse/controller/CarSaleController.java:160]: What is there wrong?
It is a web controller and as I know It uses only `get` and `post` mappings
so I chose `get` for delete operations

File: src/main/java/com/carhouse/controller/CarSaleController.java (hunk 1 above)
Comment (maksimk99) [reply src/main/java/com/carhouse/controller/CarSaleController.java]: Ok, I've understood

File: src/main/java/com/carhouse/controller/CarSaleController.java (hunk 2 above)
Comment (maksimk99) [reply src/main/java/com/carhouse/controller/CarSaleController.java]: Ok

File: src/main/java/com/carhouse/controller/CarSaleController.java [hunk 4]
```diff
@@ -90,16 +90,94 @@ public String addCarSale(final Model model) {
...
+     * Take list of selected car feature's id as request param
+     * Return page with list car sales
+     *
+     * @param carSale     the car sale
+     * @param carSaleId   the car sale id
+     * @param requestUrl  the request url
+     * @param featureList the feature list
+     * @return the string
+     */
+    @PostMapping("/carSale/{carSaleId}/update")
```
Comment (aleksieiev) [inline src/main/java/com/carhouse/controller/CarSaleController.java:141]: `/update` redundant as for me

File: src/main/java/com/carhouse/controller/CarSaleController.java [hunk 5]
```diff
@@ -90,16 +90,94 @@ public String addCarSale(final Model model) {
...
+     *
+     * @param carSaleId  the car sale id
+     * @param requestUrl the request url
+     * @return the string
+     */
+    @GetMapping("/carSale/{carSaleId}/delete")
+    public String deleteCarSale(@PathVariable final Integer carSaleId,
+                                @RequestParam("requestUrl") final String requestUrl) {
+        String redirectUrl = requestUrl.replaceAll("\\*\\*\\*", "&")
+                .replaceFirst("http://localhost:[0-9]*", "redirect:");
```
Comment (aleksieiev) [inline src/main/java/com/carhouse/controller/CarSaleController.java:164]: wtf is that? have you ever thought that this application can't work every time on localhost?

File: src/main/java/com/carhouse/controller/CarSaleController.java [hunk 6]
```diff
@@ -90,16 +90,94 @@ public String addCarSale(final Model model) {
...
+    }
+
+    /**
+     * Create array of selected car features.
+     * It is used to set selected car features when updating car sale advertisement
+     *
+     * @param carFeatureList the car feature list
+     * @return id list
+     */
+    private List<Integer> createSelectedCarFeatureList(final List<CarFeature> carFeatureList) {
```
Comment (aleksieiev) [inline src/main/java/com/carhouse/controller/CarSaleController.java:176]: 1. `stream`?
2. checks for `null`?

**Review by aleksieiev — APPROVED** [review]
Fix issues in the further PRs.

**Pull request merged**